crates/re_arrow_store/src/store.rs

```diff
@@ -97,22 +97,38 @@ impl std::ops::DerefMut for DataTypeRegistry {
 }
 
 /// Keeps track of arbitrary per-row metadata.
-#[derive(Debug, Default, Clone)]
-pub struct MetadataRegistry<T: Clone>(pub BTreeMap<RowId, T>);
+#[derive(Debug, Clone)]
+pub struct MetadataRegistry<T: Clone> {
+    pub registry: BTreeMap<RowId, T>,
+
+    /// Cached heap size, because the registry gets very, very large.
+    pub heap_size_bytes: u64,
+}
+
+impl Default for MetadataRegistry<TimePoint> {
+    fn default() -> Self {
+        let mut this = Self {
+            registry: Default::default(),
+            heap_size_bytes: 0,
+        };
+        this.heap_size_bytes = this.heap_size_bytes();
```

Comment:
seems odd? We know that this method returns zero. Not doing so would mean that the cached value can be out of date which is kinda weird.

Reply:
It's just future proofing